server/src/main/java/org/opensearch/indices/ingest/StreamPoller.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.ingest;
+
+import org.opensearch.index.IngestionShardPointer;
+
+import java.io.Closeable;
+
+/**
+ * A poller for reading messages from an ingestion shard. This is used in the ingestion engine.
+ */
+public interface StreamPoller extends Closeable {
+
+    String BATCH_START = "batch_start";
+
+    /**
+     * Start the poller
+     */
+    void start();;
```

Comment:
We do have the [LifecycleComponent](https://github.com/opensearch-project/OpenSearch/blob/78eea275a3e1812a008e815d846ca871d7f09d20/libs/common/src/main/java/org/opensearch/common/lifecycle/LifecycleComponent.java#L44) interface (and an abstract implementation). I don't know if that would be useful here but please take a look if you hadn't considered extending it.

Reply:
thanks for the suggestion. I took a look at [Lifecycle](https://github.com/opensearch-project/OpenSearch/blob/main/libs/common/src/main/java/org/opensearch/common/lifecycle/Lifecycle.java), it has a subset of state that we need here. And the state is in enum, so cannot do subclass.

also a bit different from the listener design in LifecycleComponent, the state change is fully contained within this class, and it does not have other external components attached for now. So perhaps not that useful, but I want to have your thoughts

Reply:
> So perhaps not that useful

If there's no need to use this generically as a LifecycleComponent then it's probably better to implement specifically what you need as you've done here.